core/Tracker/Model.php

```diff
@@ -418,7 +419,21 @@ public function findVisitor($idSite, $configId, $idVisitor, $userId, $fieldsToRe
 
     private function findVisitorByVisitorId($idVisitor, $select, $from, $where, $bindSql)
     {
-        // will use INDEX index_idsite_idvisitor (idsite, idvisitor)
+        $cacheKey = 'log_visit_has_index_idsite_idvisitor';
+        $cache = Cache::getCacheGeneral();
+
+        if (!array_key_exists($cacheKey, $cache)) {
```

Comment:
Can we put this into `CoreHome` plugin or some other plugin and enrich the index when  the data is gathered? Otherwise what happens is that `general cache` will be generated, then it will be regenerated here under circumstances and written twice. This becomes especially problematic when you actually want to sync the cache across servers like we do where we generate the cache every hour on one server and sync it to other servers. In this case, we would generate the cache without that field, sync it across all servers, then each server would regenerate the cache again.

Might be actually good to check if there are similar cases in the code for the cache general.

Reply:
ok. will change that. Searched for other places where the cache is written. This is the only one I found: https://github.com/matomo-org/matomo/blob/b16a791aa3650d85af829156129c2bd44c7cb075/core/Tracker/ScheduledTasksRunner.php#L65

But guess that can't be changed easily...